math/binarymath/checkisnumberpoweroftwo.go

```diff
@@ -0,0 +1,37 @@
+// checkisnumberpoweroftwo.go
+// description: Is the number a power of two
+// details:
+// Checks if a number is a power of two- [Power of two](https://en.wikipedia.org/wiki/Power_of_two)
+// author(s) [red_byte](https://github.com/i-redbyte)
+// see checkisnumberpoweroftwo_test.go
+
+package binarymath
+
+import (
+	"math"
+)
+
+func IsPowerOfTwoBinaryMethod(x int) bool {
```

Comment:
Can you choose a more representative name? `IsPowerOfTwoBinaryMethod` has actually too many things for it to be easily understandable - to the likes of someone like me, 😅.

I think the a`*BinaryMethod` can be removed from suffix and `IsPowerOfTwo` is enough for it to be clear - note `binary` is in the package name.